src/audio/host-zephyr.c

```diff
@@ -821,25 +822,27 @@ int host_common_params(struct host_data *hd, struct comp_dev *dev,
 	/* calculate DMA buffer size */
 	round_up_size = (params->frame_fmt == SOF_IPC_FRAME_S24_3LE) ? (3 * align) : align;
 	buffer_size = ROUND_UP(period_bytes, round_up_size) * period_count;
+	buffer_size_preffered = buffer_size;
 	if (hd->ipc_host.dma_buffer_size != 0)
-		buffer_size = ROUND_UP(hd->ipc_host.dma_buffer_size, buffer_size);
+		buffer_size_preffered = ROUND_UP(hd->ipc_host.dma_buffer_size, buffer_size);
 
 	/* alloc DMA buffer or change its size if exists */
 	/*
 	 * Host DMA buffer cannot be shared. So we actually don't need to lock,
 	 * but we have to write back caches after we finish anywae
 	 */
 	if (hd->dma_buffer) {
-		err = buffer_set_size(hd->dma_buffer, buffer_size, addr_align);
+		err = buffer_set_size_range(hd->dma_buffer, buffer_size_preffered, buffer_size,
+					    addr_align);
```

Comment:
also from here it seems like the case of `buffer_size_preffered == buffer_size` is quite possible